### ISSUE TITLE
I am curious about the training settings of resnet50 trained on ImageNet100

### ISSUE THREAD

**Issue by HaiyuWu** [rbalestr-lab/lejepa#29]
I am trying to reproduce the results, so I follow the config suggestions in the paper (technically not for resnet50), but I cannot reproduce the accuracy. My current highest is 82% which is 1.03% lower than this table. Hence, it will be really helpful to have the original settings from you or give me a hint which setting is wrong. : )

Thank you so much!

<img width="1162" height="272" alt="Image" src="https://github.com/user-attachments/assets/a307c205-d2db-4086-bad4-9c4bef8ea2a4" />

Config:
```
+lamb=0.05 
+lr=5e-4 
+n_global=4 
+n_local=0 
+num_proj=1024 
+global_img_size=224 
+local_img_size=98 
+proj_dim=64 
+bs=256 
+epochs=400
Three projection layers
```

Optimizers and scheduler:
```
g1 = {"params": net.parameters(), "lr": cfg.lr, "weight_decay": 5e-4}
g2 = {"params": probe.parameters(), "lr": 1e-3, "weight_decay": 1e-7}
opt = torch.optim.AdamW([g1, g2])

s1 = LinearLR(opt, start_factor=0.01, total_iters=warmup_steps)
s2 = CosineAnnealingLR(opt, T_max=total_steps - warmup_steps, eta_min=1e-3)
scheduler = SequentialLR(opt, schedulers=[s1, s2], milestones=[warmup_steps])
```
Model:
```
class ViTEncoder(nn.Module):
    def __init__(self, proj_dim=128):
        super().__init__()
        self.backbone = timm.create_model(
            "resnet50",
            pretrained=False,
            num_classes=2048,
            # drop_path_rate=0.1,
            # img_size=128,
        )
        self.proj = MLP(2048, [2048, 2048, proj_dim], norm_layer=nn.BatchNorm1d)
```
Aug:
```
self.aug = v2.Compose(
            [
                v2.RandomResizedCrop(224, scale=(0.08, 1.0)),
                v2.RandomApply([v2.ColorJitter(0.8, 0.8, 0.8, 0.2)], p=0.8),
                v2.RandomGrayscale(p=0.2),
                v2.RandomApply([v2.GaussianBlur(kernel_size=7, sigma=(0.1, 2.0))]),
                v2.RandomApply([v2.RandomSolarize(threshold=128)], p=0.2),
                v2.RandomHorizontalFlip(),
                v2.ToImage(),
                v2.ToDtype(torch.float32, scale=True),
                v2.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
            ]
        )
        self.test = v2.Compose(
            [
                v2.Resize(256),
                v2.CenterCrop(224),
                v2.ToImage(),
                v2.ToDtype(torch.float32, scale=True),
                v2.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
            ]
        )
```

**Comment by RandallBalestriero** [comment]
Thanks for your interest! A few small differences between the particular setup of that table (not sure if they would help or not):
- our test transform uses resize(224) and  centercrop(224)
- for the Gaussian blur we actually use a PIL version (same as used e.g. in BarlowTwins repo etc)
- colorjittering was 0.4 0.4 0.2 0.1 parameters
- we used greater projector output dimension but not sure this would impact performances
- perhaps you can try with 6 or 8 views

Otherwise things look good to me!

**Comment by HaiyuWu** [comment]
Thanks for your reply! I have made these changes, let's see what result I can get. : )
From my ablation study and the paper's, dimension does impact the accuracy but very small when they are not too small.
